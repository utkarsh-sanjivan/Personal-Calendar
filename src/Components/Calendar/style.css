.calendar-container {
    border: 1px black solid;
}

.today-date-card {
    border: 2px blue solid;
    width: 14.2857143%;
    height: 140px;
}

.date-card {
    width: 14.2857143%;
    height: 140px;
}

.blocked-date-card {
    background-color: grey;
    width: 14.2857143%;
    height: 140px;
}