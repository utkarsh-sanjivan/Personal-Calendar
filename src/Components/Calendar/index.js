import React from 'react';
import DateCard from '../DateCard/index';
import { Row } from 'antd';
import './style.css';

const Calendar = props => (
    <div className='calendar-container'>
        <Row>
            {['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'].map(day=> <DateCard day={day}/>)}
        </Row>
        {props.dateColumn.map(dateArr => (
            <Row>
                {dateArr.map(date => 
                    <DateCard
                        onDateClick={selectedDay => { if(date.isActive) props.onDateClick(selectedDay) }}
                        onSeeMoreClick={selectedDay => props.onSeeMoreClick(selectedDay)}
                        onEventClick={selectedEvent => props.onEventClick(selectedEvent)}
                        formattedDate={date.formattedDate}
                        isActive={date.isActive}
                        events={date.events}
                        date={date}
                        className={
                            date.isActive?
                                date.isToday? 
                                    'today-date-card'
                                    : 'date-card'
                                : 'blocked-date-card'
                        }
                    />
                )}
            </Row>
        ))}
    </div>
)

export default Calendar;
