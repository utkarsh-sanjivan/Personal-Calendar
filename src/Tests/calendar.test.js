import React from 'react';
import { create } from "react-test-renderer";
import Calendar from './../Components/Calendar/index';

test('Test Calendar Component', () => {
const component = create(<Calendar 
  dateColumn={[]}
  currentDate={''}
  onDateClick={()=>{}}
  onSeeMoreClick={()=>{}}
  onEventClick={()=>{}}
  />);
  expect(component.toJSON()).toMatchSnapshot()
});