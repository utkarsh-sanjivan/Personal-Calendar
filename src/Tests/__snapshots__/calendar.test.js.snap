// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calendar Component 1`] = `
<div
  className="calendar-container"
>
  <div
    className="ant-row calendar-days-row"
    style={Object {}}
  >
    <div
      className="card-container"
    >
      <div
        className="day-card"
      >
        Sunday
      </div>
    </div>
    <div
      className="card-container"
    >
      <div
        className="day-card"
      >
        Monday
      </div>
    </div>
    <div
      className="card-container"
    >
      <div
        className="day-card"
      >
        Tuesday
      </div>
    </div>
    <div
      className="card-container"
    >
      <div
        className="day-card"
      >
        Wednesday
      </div>
    </div>
    <div
      className="card-container"
    >
      <div
        className="day-card"
      >
        Thursday
      </div>
    </div>
    <div
      className="card-container"
    >
      <div
        className="day-card"
      >
        Friday
      </div>
    </div>
    <div
      className="card-container"
    >
      <div
        className="day-card"
      >
        Saturday
      </div>
    </div>
  </div>
</div>
`;
